cfg/builder/builder_entry.cc

```diff
@@ -79,8 +79,8 @@ unique_ptr<CFG> CFGBuilder::buildFor(core::Context ctx, ast::MethodDef &md) {
     dealias(ctx, *res);
     CFG::ReadsAndWrites RnW = res->findAllReadsAndWrites(ctx);
     computeMinMaxLoops(ctx, RnW, *res);
-    removeDeadAssigns(ctx, RnW, *res);
     fillInBlockArguments(ctx, RnW, *res);
+    removeDeadAssigns(ctx, RnW, *res); // requires block arguments to be filled
```

Comment:
small note, it took me a second to realize that this meant "require basic block arguments to be filled"

after looking at the implementation of `removeDeadAssigns` it's clear that what this means, but we might want to rename `fillInBlockArguments` to `fillInBasicBlockArguments`.

Reply:
(non blocking)